frontend/src/App.tsx

```diff
@@ -97,17 +98,28 @@ export default class App extends React.Component<{}, State> {
 
     this.state.nodes.setComparator(this.getComparator(this.sortBy.get()));
 
-    this.connection = Connection.create(this.pins, (changes) => {
-      if (changes) {
-        this.setState(changes);
-      }
-
-      return this.state;
-    });
+    this.connection = Connection.create(this.pins, this.setStateLazy);
 
     setInterval(() => (this.chainsCache = []), 10000); // Wipe sorted chains cache every 10 seconds
   }
 
+  private setStateLazy: Update = (changes) => {
+    // Apply new changes to the state immediately
+    Object.assign(this.state, changes);
+
+    // Trigger React update on next animation frame only once
+    if (!this.isUpading) {
```

Comment:
React batches DOM updates via VDOM diffing, but it will still trigger separate VDOM diffs per `setState`. VDOM is faster than DOM, but you still have to crawl down a tree (two trees in fact) of arbitrarily allocated pointers, which is one of the worst things you can do to CPU cache. This was already sort of done in [Connection.ts](https://github.com/paritytech/substrate-telemetry/blob/675776c3e13fff603d3f17604f88ea559606124c/frontend/src/Connection.ts#L419-L425), this just moves the logic so that it can affect all updates.